src/tuncer.erl

```diff
@@ -376,6 +376,27 @@ set_mode(passive, Port) ->
             {error, Error}
     end.
 
+-spec up_tunnel(fun(), list()) -> ok | {error, term()}.
+up_tunnel(UpCallback, Flag) ->
+    Extra = proplists:get_value(extra, Flag, []),
+    case proplists:get_value(namespace, Extra) of
+        undefined ->
+            UpCallback();
+        NS ->
+            up_into_namespace(UpCallback, NS)
+    end.
+
+up_into_namespace(UpCallback, NS) ->
+    case procket:open_nif("/proc/self/ns/net", read) of
```

Comment:
Any linux specific code should be moved to tunctl_linux.erl.

With your changes to procket, procket:dev/2 returns an fd to the tuntap device in the namespace. This function:
* gets a fd to the current namespace
* moves the beam thread we're scheduled on to a new namespace
* calls ioctl() to configure the tuntap address
* moves the beam thread back to the original namespace

This could potentially be racy (another function using the network is run).